client/src/pages/ManageBookings/ManageBookings.tsx

```diff
@@ -16,39 +17,33 @@ import { useHistory } from 'react-router-dom';
 import useStyles from './useStyles';
 
 const ManageBookings = (): JSX.Element => {
-  const { loggedInUser } = useAuth();
   const { initSocket } = useSocket();
 
-  const history = useHistory();
   const classes = useStyles();
 
   useEffect(() => {
     initSocket();
   }, [initSocket]);
 
-  const [date, changeDate] = useState<MaterialUiPickersDate>(new Date());
+  const [date, setDate] = useState<MaterialUiPickersDate>(new Date());
   const [isSelectedDate, setIsSelectedDate] = useState<boolean>(false);
 
   const [requests, setRequests] = useState<Array<BookingRequest>>(mockBookingRequest);
-  const nextBooking = requests.reduce((request, closest) =>
-    new Date(request.start.getTime() - new Date().getTime()) < closest.start &&
-    request.start &&
-    request.start > new Date()
-      ? (closest = request)
-      : (closest = closest),
-  );
-  const [selectedBooking, setSelectedBooking] = useState<BookingRequest | undefined>(nextBooking);
 
-  if (loggedInUser === undefined) return <CircularProgress />;
-  if (!loggedInUser) {
-    history.push('/login');
-    // loading for a split seconds until history.push works
-    return <CircularProgress />;
-  }
+  const nextBooking = requests
+    .filter((request) => request.start.getTime() - new Date().getTime() > 0)
+    .reduce(
+      (request, closest) =>
+        closest.start.getTime() - new Date().getTime() > request.start.getTime() - new Date().getTime()
+          ? (closest = request)
+          : (closest = closest),
+      requests[0],
```

Comment:
I made a mistake in saying that we should do this since reduce docs state:
> If initialValue is not specified, previousValue is initialized to the first value in the array, and currentValue is initialized to the second value in the array.

Hence, no need to do this explicitly. 👍🏼